app/models/live_result.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class LiveResult < ApplicationRecord
+  has_many :live_attempts, -> { where(replaced_by: nil).order(:attempt_number) }
+
+  after_save :notify_users
+
+  belongs_to :registration
+
+  belongs_to :entered_by, class_name: 'User', foreign_key: 'entered_by_id'
+
+  belongs_to :round
+
+  has_one :event, through: :round
+
+  DEFAULT_SERIALIZE_OPTIONS = {
+    only: %w[ranking registration_id round live_attempts round best average single_record_tag average_record_tag advancing advancing_questionable entered_at entered_by_id],
+    methods: %w[event_id attempts result_id],
+    include: %w[event live_attempts round],
```

Comment:
This is a heck of a lot of data you're serializing for each and every single scorecard. I won't stand in your way for your PoC experiments, but please think about how to slim this down. Especially for `event`, just sending along `event_id` and accessing the static list of WCA events in JS should be enough.

For the round, I'd also argue that a round ID should do, and you keep a lookup somewhere else in your React component.